File: frontend/src/services/clientService.js
import api from "./api";

export const getClients = () => api.get("/clients");

export const createClient = (client) =>
  api.post("/clients", client);

export const updateClient = (id, client) =>
  api.put(`/clients/${id}`, client);

export const deleteClient = (id) =>
  api.delete(`/clients/${id}`);

File: backend/service-clients/src/main/java/org/ateliers/serviceclients/Services/ClientService.java
package org.ateliers.serviceclients.Services;

import org.ateliers.serviceclients.DTOs.ClientRequestDTO;
import org.ateliers.serviceclients.DTOs.ClientResponseDTO;
import java.util.List;

public interface ClientService {
    // 1. GET all
    List<ClientResponseDTO> getAllClients();

    // 2. GET by ID
    ClientResponseDTO getClientById(Long id);

    // 3. POST create
    ClientResponseDTO createClient(ClientRequestDTO clientRequestDTO);

    // 4. PUT update
    ClientResponseDTO updateClient(Long id, ClientRequestDTO clientRequestDTO);

    // 5. DELETE
    void deleteClient(Long id);
}